False positives with keywords-only fields of dataclasses

### Bug description

```python
from dataclasses import dataclass, KW_ONLY

@dataclass
class C:
    _: KW_ONLY
    c: int

c = C(c=1)  #  No value for argument '_' in constructor call (no-value-for-parameter)
```
### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.0 (default, Nov  6 2021, 07:41:13) [Clang 12.0.5 (clang-1205.0.22.9)]
```

Happens with the `kw_only` attribute of `dataclasses.field` too:
```python
"""Quick test to demonstrate pylint errors when using kw_only dataclass
fields."""
from dataclasses import dataclass, field

@dataclass
class Creature:
    """Defines a creature with one or more names (first names, middle names,
    last names, etc.)"""
    names: list[str] = field(kw_only=True)

@dataclass
class Human(Creature):
    """Defines a human who has a home address."""
    address: str

donna = Human('1858 Hidden Meadow Drive, East Grand Forks, ND 56721',
              names=['Donna', 'Jennings'])
print(donna)
```

Running pylint (2.12.2):
```
$ python -m pylint dc_kw_only.py 
************* Module dc_kw_only
dc_kw_only.py:16:8: E1124: Argument 'names' passed by position and keyword in constructor call (redundant-keyword-arg)
dc_kw_only.py:16:8: E1120: No value for argument 'address' in constructor call (no-value-for-parameter)

--------------------------------------------------------------------
Your code has been rated at -4.29/10
```

--- a/doc/whatsnew/fragments/5767.false_positive
+++ b/doc/whatsnew/fragments/5767.false_positive
@@ -1,0 +1,3 @@
+Don't report ``no-value-for-parameter`` for dataclasses fields annotated with ``KW_ONLY``.
+
+Closes #5767

--- a/tests/functional/d/dataclass_parameter.py
+++ b/tests/functional/d/dataclass_parameter.py
@@ -1,0 +1,27 @@
+"""Tests for dataclass and checks that check for parameters."""
+
+import dataclasses
+from dataclasses import KW_ONLY as keyword_only
+from dataclasses import dataclass
+
+
+@dataclass
+class MyDataClass:
+    """Simple dataclass with a KW_ONLY parameter."""
+
+    _: dataclasses.KW_ONLY
+    data: str
+
+
+MyDataClass(data="test")
+
+
+@dataclass
+class MyDataClassWithAliases:
+    """Simple dataclass with an aliased KW_ONLY parameter."""
+
+    _: keyword_only
+    data: str
+
+
+MyDataClassWithAliases(data="test")

--- a/tests/functional/d/dataclass_parameter.rc
+++ b/tests/functional/d/dataclass_parameter.rc
@@ -1,0 +1,2 @@
+[testoptions]
+min_pyver=3.10

